tests/integrationv2/common.py

```diff
@@ -35,29 +57,109 @@ def __str__(self):
+
+    def __gt__(self, other):
+        return self.value > other.value
+
+    def __lt__(self, other):
+        return self.value < other.value
+
+    def __eq__(self, other):
+        return self.value == other.value
+
+    def __str__(self):
+        return self.name
+
+
+class Protocols(object):
     """
-    When referencing curves, use these class values.
-    Don't hardcode curve names.
+    When referencing protocols, use these protocol values.
     """
-    X25519 = "X25519"
-    P256 = "P-256"
-    P384 = "P-384"
+    TLS13 = Protocol("TLS1.3", 34)
+    TLS12 = Protocol("TLS1.2", 33)
+    TLS11 = Protocol("TLS1.1", 32)
+    TLS10 = Protocol("TLS1.0", 31)
+    SSLv3 = Protocol("SSLv3", 30)
+
+
+class Cipher(object):
+    def __init__(self, name, min_version, compat, compat2):
+        self.name = name
+        self.min_version = min_version
+        self.compat = compat
 
+    def __eq__(self, other):
+        return self.name == other
 
-class Ciphersuites(object):
+    def __hash__(self):
+        return hash(self.name)
+
+    def __str__(self):
+        return self.name
+
+
+class Ciphers(object):
+    """
+    When referencing ciphers, use these class values.
     """
-    When referencing ciphersuites, use these class values.
-    Don't hardcode the the ciphersuite names.
+    RC4_MD5 = Cipher("RC4_MD5", Protocols.SSLv3, False, False)
+    RC4_SHA = Cipher("RC4_SHA", Protocols.SSLv3, False, False)
+    DES_CBC3_SHA = Cipher("DES_CBC3_SHA", Protocols.SSLv3, False, True)
+    DHE_RSA_DES_CBC3_SHA = Cipher("DHE_RSA_DES_CBC3_SHA", Protocols.SSLv3, False, False)
+    AES128_SHA = Cipher("AES128_SHA", Protocols.SSLv3, True, True)
+    DHE_RSA_AES128_SHA = Cipher("DHE_RSA_AES128_SHA", Protocols.SSLv3, True, False)
+    AES256_SHA = Cipher("AES256_SHA", Protocols.SSLv3, True, True)
+    DHE_RSA_AES256_SHA = Cipher("DHE_RSA_AES256_SHA", Protocols.SSLv3, True, False)
+    AES128_SHA256 = Cipher("AES128_SHA256", Protocols.TLS12, True, True)
+    AES256_SHA256 = Cipher("AES256_SHA256", Protocols.TLS12, True, True)
+    DHE_RSA_AES128_SHA256 = Cipher("DHE_RSA_AES128_SHA256", Protocols.TLS12, True, True)
+    DHE_RSA_AES256_SHA256 = Cipher("DHE_RSA_AES256_SHA256", Protocols.TLS12, True, True)
+    AES128_GCM_SHA256 = Cipher("AES128_GCM_SHA256", Protocols.TLS13, True, True)
+    AES256_GCM_SHA384 = Cipher("AES256_GCM_SHA384", Protocols.TLS13, True, True)
+    DHE_RSA_AES128_GCM_SHA256 = Cipher("DHE_RSA_AES128_GCM_SHA256", Protocols.TLS12, True, True)
+    DHE_RSA_AES256_GCM_SHA384 = Cipher("DHE_RSA_AES256_GCM_SHA384", Protocols.TLS12, True, True)
+    #ECDHE_ECDSA_AES128_SHA = Cipher("ECDHE_ECDSA_AES128_SHA", Protocols.SSLv3, True, False)
+    #ECDHE_ECDSA_AES256_SHA = Cipher("ECDHE_ECDSA_AES256_SHA", Protocols.SSLv3, True, False)
+    #ECDHE_ECDSA_AES128_SHA256 = Cipher("ECDHE_ECDSA_AES128_SHA256", Protocols.TLS12, True, True)
+    #ECDHE_ECDSA_AES256_SHA384 = Cipher("ECDHE_ECDSA_AES256_SHA384", Protocols.TLS12, True, True)
```

Comment:
Q: Why are these ciphers disabled at the moment?

Reply:
I've added several more Ciphers in the latest push. These are included.